### DISCUSSION TITLE
Working with synthetics rocks generated from a DEM.

### DISCUSSION THREAD

**Discussion by MonkeyLever** [PMEAL/porespy#571] [Q&A]
I am learning digital rock physics. I'm generating a synthetic rock using a particle pack generator (https://github.com/MosGeo/ParticlePack#particula---particle-pack-generator). I'm wondering what is the ideal workflow to generate work with the synthetic rock. My idea is to generate the synthetic and then use porespy and openpnm to get familiar with the workflow.
Any suggestions are welcome.
Rgds.

**Comment by jgostick** [comment]
You don't make it clear in your question about your current state of knowledge, so pardon me if the following is already known...

Almost all 'digital rock' tools require having the image as 3D tiff image.  This is a single file in the tiff format that has all the 'layers' in the same file.  You can open these images in ImageJ as a 'stack' and scroll through the layers.  I will recommend the [youtube tutorial series](https://www.youtube.com/watch?v=zqFISBryLxk) by @ArashRabbani as a good place to get started.  They use matlab while our tools use python, but these are generally similar.

Once you have the 3D tiff image you can open it in python and work on it just like it's a 3D array of integers. 1's would be the void and 0's the solid (or more numbers if you have more phases).  You can pass this array into any function like the filters in porespy, then get an array back.  You can 'analyze' the result for instance by doing histograms of the values, etc.  

Anyway, hopefully that's the sort of info you were looking for.  Good luck!

**Reply by MonkeyLever** [reply]
Thanks for your response. My current state of knowledge is very basic. 

Thanks for the suggestions I have been reviewing the youtube tutorial that you mention.

Thanks again.
